tests/e2e/bcd_consumer_integration_test.go

```diff
@@ -548,14 +550,86 @@ func (s *BCDConsumerIntegrationTestSuite) Test09ConsumerFPFinalitySignature() {
 	s.True(finalizedBlock.Finalized)
 }
 
-// Test10BabylonFPCascadedSlashing
+func (s *BCDConsumerIntegrationTestSuite) Test10BsnFpRewards() {
+	// Ensure BSN rewards are being gathered.
+	// Initially sent to the finality contract, then sent to Babylon on an epoch boundary
+	s.Eventually(func() bool {
+		balance, err := s.cosmwasmController.QueryFinalityContractBalances()
+		if err != nil {
+			s.T().Logf("failed to query balance: %s", err.Error())
+			return false
+		}
+		if len(balance) == 0 {
+			return false
+		}
+		positiveBalance := false
+		for _, b := range balance {
+			denom := b.Denom
+			s.T().Logf("Balance of denom '%s': %s", denom, b.Amount.String())
+			// Check that the balance of the denom is greater than 0
+			if b.Amount.IsPositive() {
+				positiveBalance = true
+			} else {
+				s.T().Logf("balance of denom '%s' is not positive: %s", denom, b.Amount.String())
+			}
+		}
+		return positiveBalance
+	}, 30*time.Second, time.Second*5)
+
+	// Wait for an epoch boundary on the BSN (reward_interval = 50, so BSN height has to be a multiple of 50)
+	s.T().Logf("Waiting for an epoch boundary on the BSN")
+	rewardInterval := uint64(50)
```

Comment:
Maybe better to get the interval from finality config? We can also change it to a small value if needed to execute faster. Actually, I'm not sure adding this wait time make sense because the rewards are accumulated before this wait